server/src/main/java/org/elasticsearch/cluster/routing/UnassignedInfo.java

```diff
@@ -67,7 +67,7 @@ public record UnassignedInfo(
     Reason reason,
     @Nullable String message,
     @Nullable Exception failure,
-    int failedAllocations,
+    int failureCount,
```

Comment:
I prefer to keep existing name. I think it is more specific this way (failures to allocate) not just any generic failure

Reply:
I found failedAllocations as integer confusing, sounds like it should be a collection of some sort of allocation objects. Also there is "failure" field, which I believe should be "allocationFailure", isnt it?